tiled/catalog/adapter.py

```diff
@@ -1479,6 +1480,19 @@ def in_or_not_in(query, tree, method):
     return _IN_OR_NOT_IN_DIALECT_DISPATCH[dialect_name](query, tree, method)
 
 
+def has_key(query, tree):
+    # Functionally in SQLAlchemy 'is not None' does not work as expected
+    if tree.engine.url.get_dialect().name == "sqlite":
```

Comment:
Due to changes in `main`, this now needs to be `tree.context.engine`. That should address the new test failures.